app/services/group/seed.rb

```diff
@@ -0,0 +1,52 @@
+module Group
+  class Seed
+    attr_reader :status
+    def initialize(tenant)
+      @user = ManageIQ::API::Common::Request.current.user
+      validate(tenant)
+    end
+
+    def validate(tenant)
+      account_number = ManageIQ::API::Common::Request.current.identity['identity']['account_number']
+      raise Catalog::NotAuthorized if account_number != tenant.external_tenant
+      raise Catalog::NotAuthorized unless @user.org_admin?
+
+      true
+    end
+
+    def tenant
+      @user.tenant
+    end
+
+    def code(status)
+      @status = status
+    end
+
+    def process
+      lookup_groups
+      cache_tenant
+      @seeded.data.present? ? code(204) : run_seeding
+      self
+    end
+
+    private
+
+    def run_seeding
+      seeded = RBAC::Seed.new(Rails.root.join('data', 'rbac_catalog_seed.yml')).process
```

Comment:
@syncrou should we allow this seeding to happen for multiple files or just a single file.

Reply:
@mkanoor I'm not sure it that would be helpful or not. Are you anticipating that we would have mutliple seeding files in the future?